### PR TITLE
Avoid injected 'observe' & 'transaction' dependencies persist to 'Dependencies.toml'

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29372

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java [hunk 1]
```diff
@@ -520,35 +519,64 @@ private static boolean validateDotSeparatedIdentifiers(String identifiers) {
...
+    private static boolean isPkgManifestDependency(ResolvedPackageDependency graphDependency,
+            List<PackageManifest.Dependency> pkgManifestDependencies) {
+        if (pkgManifestDependencies.isEmpty()) {
+            return false;
+        }
+
+        for (PackageManifest.Dependency manifestDependency : pkgManifestDependencies) {
+            if (manifestDependency.org().value().equals(graphDependency.packageInstance().packageOrg().value())
+                    && manifestDependency.name().value()
+                    .equals(graphDependency.packageInstance().packageName().value())) {
```
Comment (lnash94) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:566]: Don't we need to trim white space in string comparison? 
`manifestDependency.name().value().trim().equals(graphDependency.packageInstance().packageName().value().trim())`

Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:566]: I'm not sure why we need to do that, Were do you found issues by comparing these?

Comment (lnash94) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:566]: yeah, this won't need, if we don't allow white space in packageName. :)

Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:566]:
```

Comment (lnash94) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:566]: yeah sure :)

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]
Can we not use boolean to represent this? I think we should use an enum.

**Review by hevayo — DISMISSED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/DependencyResolvedType.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *  under the License.
+ */
+package io.ballerina.projects;
+
+/**
+ * Represents the type of the package dependency resolution.
+ *
+ * @since 2.0.0
+ */
+public enum DependencyResolvedType {
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/DependencyResolvedType.java:25]: Better name would be DependancyResolutionType
SOURCE
INJECTED

Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/DependencyResolvedType.java]: resolved in b47a264

**Review by hevayo — APPROVED** [review]

**Review by sameerajayasoma — APPROVED** [review]
LGTM

**Pull request merged**